app/models/registry.rb

```diff
@@ -1,8 +1,6 @@
-# model that represents a docker registry
-class DockerRegistry < ActiveRecord::Base
-  has_one :certificate_service, as: :service, dependent: :destroy
-  has_one :certificate, through: :certificate_service
-
+# model that represents a registry
+class Registry < ActiveRecord::Base
+  validates :name, presence: true, uniqueness: true
```

Comment:
This should `have_many :mirrors`, no?

Reply:
It should also have:

```
has_one :certificate_service, as: :service, dependent: :destroy
has_one :certificate, through: :certificate_service
```

As one thing is the certificate the registry has, and another  one the certificate the mirror has.